streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1109,7 +1109,6 @@ public void onAssignment(final Assignment assignment, final ConsumerGroupMetadat
         // Check if this was a version probing rebalance and check the error code to trigger another rebalance if so
         if (maybeUpdateSubscriptionVersion(receivedAssignmentMetadataVersion, latestCommonlySupportedVersion)) {
             setAssignmentErrorCode(AssignorError.VERSION_PROBING.code());
-            return;
```

Comment:
this is the actual fix: it needs to be applied to both the real assignor and the VP system test's assignor to get a green. We should consolidate these so we don't keep forgetting to add a fix to the test's custom assignor, and thinking it's snot actually fixed